DataScience/GpxWavLabeler.py

```diff
@@ -0,0 +1,501 @@
+
+                # Identify valid time sets in which succusive times
+                # and no more than the specified delta time
+                dub_t = vld_t[(pos_plt_idx | neg_plt_idx) & dot_plt_idx & spd_plt_idx]
+                dub_t_sets = np.split(dub_t, np.where(np.diff(dub_t) > DELTA_T_MAX)[0] + 1)
+
+                # Export the first clip having at least two valid
+                # times
+                for dub_t_set in dub_t_sets:
+                    if dub_t_set.shape[0] > 2:
+                        start_t = int(dub_t_set[0] * 1000)
+                        stop_t = int(dub_t_set[-1] * 1000)
+                        clip = audio[start_t:stop_t]
+                        wav_filename = "clip{:+.1f}{:+.1f}"
+                        if isgreater:
+                            wav_filename += "-greater"
+                        else:
+                            wav_filename += "-lesser"
+                        wav_filename += "{:+.1f}.wav"
+                        clip.export(
+                            CLIP_HOME / wav_filename.format(
+                                heading_centers[pos_lbl_idx][0],
+                                heading_centers[neg_lbl_idx][0],
+                                speed_centers[spd_lbl_idx][0],
+                                ), format="wav")
+                        break
+
+                # Optionally plot the track and color points
+                # corresponding to the current heading and speed
+                # cluster centers
+                if do_plot:
+                    fig, axs = plt.subplots()
+                    axs.plot(r_s_o[0, :], r_s_o[1, :])
+                    axs.plot(
+                        r_s_o[0, pos_plt_idx & dot_plt_idx & spd_plt_idx],
+                        r_s_o[1, pos_plt_idx & dot_plt_idx & spd_plt_idx],
+                        ".",
+                    )
+                    axs.plot(
+                        r_s_o[0, neg_plt_idx & dot_plt_idx & spd_plt_idx],
+                        r_s_o[1, neg_plt_idx & dot_plt_idx & spd_plt_idx],
+                        ".",
+                    )
+                    title = "headings = {:.1f}, {:.1f} deg"
+                    if isgreater:
+                        title += ", heading_dot > 1"
+                    else:
+                        title += ", heading_dot < 1"
+                    title += ", speed = {:.1f} m/s"
+                    axs.set_title(
+                        title.format(
+                            heading_centers[pos_lbl_idx][0],
+                            heading_centers[neg_lbl_idx][0],
+                            speed_centers[spd_lbl_idx][0],
+                        )
+                    )
+                    axs.set_xlabel("east [m]")
+                    axs.set_ylabel("north [m]")
+                    plt.show()
+
+
+"""Demonstrate GpxLabeler.
+"""
+if __name__ == "__main__":
+
+    gpx_filename = "suuntoapp-Motorsports-2022-02-22T18-09-01Z-track.gpx"
```

Comment:
Would it be possible to make these into cmd parameters? argc/argv is okay unless ArgumentParser is trivial

Reply:
Sure.